RecoLocalTracker/SiPixelRecHits/interface/SiPixelTemplate2D.h

```diff
@@ -134,35 +174,62 @@ class SiPixelTemplate2D {
    float qscale() {return qscale_;}    //!< charge scaling factor
    float s50() {return s50_;}          //!< 1/2 of the pixel threshold signal in adc units
    float sxymax() {return sxymax_;}    //!< max pixel signal for pixel error calculation
-   float xytemp(int j, int i) {        //!< get the 2-d template for pixel j (x), i (y) in BXM2 x BYM2 array (x,y)=(0,0) in lower LH corner of pixel[1][1]
+   float scalex(int i) {
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
-      if(j < 0 || j > BXM3 || i < 0 || i > BYM3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::xytemp called with illegal indices = " << j << "," << i << std::endl;}
+      if(i < 0 || i > 3) {throw cms::Exception("DataCorrupt") << "SiPixelTemplate2D::scalex called with illegal index = " << i << std::endl;}
 #else
-      assert((j>=0 && j<BYM2) && (i>=0 && i<BYM2));
+      assert(i>=0 && i<4);
```

Comment:
constants are replaced by magic numbers is not good

Reply:
Not sure, I need to ask Morris.  (And he is not around today.)

Reply:
I propose we leave this for later.

Reply:
These constants, now disregarded, are taken from RecoLocalTracker/SiPixelRecHits/interface/SiPixelTemplateDefs.h
When you'll restore them, please also consider my previous comment https://github.com/cms-sw/cmssw/pull/22458/files#r173614394